### PR TITLE
[gettext] Remove `SUBPATH`, add iconv linking info

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  - [x] Replaces the `SUBPATH` argument to `vcpkg_build_make`, as discussed in  #19753 (CC @JackBoosY)
  - [x] Adds `Iconv_LIBRARIES` to `Intl_LIBRARIES` in Intl wrapper, as discussed in #19195 (CC @wrobelda)
    ~Reverts the kf5i18n patching for the gettext -> iconv dependency which is now handled in gettext (https://github.com/microsoft/vcpkg/pull/20090#issuecomment-919840095).~ (kf5i18n doesn't use `FindIntl.cmake`.)
  - [x] Fixes mingw-dynamic builds (from upstream).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no
  Tested locally in x64-mingw-static cross build.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

File: ports/gettext/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -7,4 +7,8 @@ if(Intl_FOUND AND Intl_LIBRARIES)
     unset(Intl_LIBRARIES)
     unset(Intl_LIBRARIES CACHE)
     select_library_configurations(Intl)
-endif()
\ No newline at end of file
+    find_package(Iconv) # Since CMake 3.11
+    if(Iconv_FOUND AND NOT Iconv_IS_BUILT_IN)
+        list(APPEND Intl_LIBRARIES ${Iconv_LIBRARIES})
```
Comment (JackBoosY) [inline ports/gettext/vcpkg-cmake-wrapper.cmake:12]: Should we fix libintl or add a cmake wrapper to add iconv libraries to `Intl_LIBRARIES`?

**Review by JackBoosY — APPROVED** [review]

**Comment by wrobelda** [comment]
@dg0yt can you also remove the `libiconv` patch from `kf5i18n`? This would test this PR nicely.

**Comment by dg0yt** [comment]
> @dg0yt can you also remove the `libiconv` patch from `kf5i18n`? It would test this PR nicely, given that many other `kf5` ports depend on `libiconv` in `kf5i18n`, see e.g. [#19195 (comment)](https://github.com/microsoft/vcpkg/pull/19195#issuecomment-916859938) or [#19766 (comment)](https://github.com/microsoft/vcpkg/pull/19766#issuecomment-916857477).

@wrobelda I have to revert the related change. The issue is that kf5i18n doesn't use the wrapped `FindIntl.cmake` but maintains its own `FindLibIntl.cmake`.

**Review by JackBoosY — APPROVED** [review]

File: ports/gettext/vcpkg-cmake-wrapper.cmake [hunk 2]
```diff
@@ -7,4 +7,8 @@ if(Intl_FOUND AND Intl_LIBRARIES)
     unset(Intl_LIBRARIES)
     unset(Intl_LIBRARIES CACHE)
     select_library_configurations(Intl)
-endif()
\ No newline at end of file
+    find_package(Iconv) # Since CMake 3.11
```
Comment (BillyONeal) [inline ports/gettext/vcpkg-cmake-wrapper.cmake:10]: Should this use `find_dependency`?

Comment (ras0219-msft) [reply ports/gettext/vcpkg-cmake-wrapper.cmake:10]: It can't, because vcpkg-cmake-wrapper acts like a find module.

**Comment by BillyONeal** [comment]
Thanks for the bugfix!

File: ports/gettext/vcpkg-cmake-wrapper.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/gettext/vcpkg-cmake-wrapper.cmake:10]: Oh, I should finish the "tribal knowledge" wrapper documentation PR, really. 😄

Comment (BillyONeal) [reply ports/gettext/vcpkg-cmake-wrapper.cmake:10]: @dg0yt This is less "tribal knowledge" and more "Bill doesn't really understand CMake as well as he should"

**Pull request merged**